sdk/storage/azure-storage-files-datalake/src/datalake_directory_client.cpp

```diff
@@ -198,36 +198,61 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     return DataLakePathClient::DeleteIfExists(deleteOptions, context);
   }
 
-  Azure::Response<Models::ListPathsSinglePageResult> DataLakeDirectoryClient::ListPathsSinglePage(
+  ListPathsPagedResponse DataLakeDirectoryClient::ListPaths(
       bool recursive,
-      const ListPathsSinglePageOptions& options,
+      const ListPathsOptions& options,
       const Azure::Core::Context& context) const
   {
     _detail::DataLakeRestClient::FileSystem::ListPathsOptions protocolLayerOptions;
     protocolLayerOptions.Resource = _detail::FileSystemResourceType::Filesystem;
     protocolLayerOptions.Upn = options.UserPrincipalName;
-    protocolLayerOptions.ContinuationToken = options.ContinuationToken;
+
     protocolLayerOptions.MaxResults = options.PageSizeHint;
     protocolLayerOptions.RecursiveRequired = recursive;
-    auto currentPath = m_pathUrl.GetPath();
-    // Remove the filesystem name and get directory name.
-    auto firstSlashPos = currentPath.find_first_of("/");
 
-    if (firstSlashPos == 0 || (firstSlashPos == currentPath.size() + 1U))
+    const std::string currentPath = m_pathUrl.GetPath();
+    auto firstSlashPos = std::find(currentPath.begin(), currentPath.end(), '/');
+    const std::string fileSystemName(currentPath.begin(), firstSlashPos);
+    if (firstSlashPos != currentPath.end())
```

Comment:
nit: Was this logic change necessary for this PR? In the future, consider making such changes in separate PRs rather than trying to introduce unrelated changes in the same one (particularly when test coverage isn't 90%+).